Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -111,6 +116,18 @@ ILoggerFactory loggerFactory
         notificationPublisher.SendStatusNotification(textDocument, CompilationStatus.ParsingFailed);
         return CreateDocumentWithEmptySymbolTable(loggerFactory.CreateLogger<SymbolTable>(), textDocument, errorReporter, program, loadCanceled: false);
       }
+
+      // We load plugins at most once.
+      if (dafnyPluginsOptions.Value.Plugins.Length > 0 && !DafnyOptions.O.Plugins.Any()) {
+        try {
+          foreach (var pluginPathArgument in dafnyPluginsOptions.Value.Plugins) {
+            DafnyOptions.O.Parse(new[] { "-plugin:" + pluginPathArgument });
+          }
+        } catch (Exception e) {
+          errorReporter.Error(MessageSource.Parser, program.GetFirstTopLevelToken(), "Error while instantiating the plugins:" + e.ToString());
```

Comment:
Maybe log this error?
Calling `ToString()` explicitly should be unnecessary in string-concatenations.

Reply:
One unrelated problem is that, by moving this code to the initialization, I don't have the reporter anymore to report to the IDE that plugin loading failed though. In the newly moved code, I am using a logger so I can just pass the exception as the first parameter.

Reply:
That's a pity. I think it'd be useful to somehow display such errors within the IDE. But I suggest leaving this to a separate PR.

Reply:
I did not follow your suggestion and implemented this mechanism right into this PR. Basically, for every load request, after parsing the program, it will fail if there were any errors during loading. I do so because I need a valid token of the program to be able to report the error in the diagnostics window. It works like a charm. See my last commit.